### ISSUE TITLE
[Bug]: file_get_contents in LibraryProperties triggers WP Plugin Check

### ISSUE THREAD

**Issue by Biont** [inpsyde/modularity#50]
### Description of the bug

This is a "bug" more in the way of an "annoyance" rather than _a problem with the code_.

When a plugin is tested against the official **Plugin Check** utility, one of the warnings/errors will be that we're supposed to use `wp_remote_get` instead of `file_get_contents`. This is clearly a false-positive since we are not making an HTTP call.

There has been an [attempt](https://github.com/WordPress/WordPress-Coding-Standards/pull/1374/files#diff-400e43bc09c24262b43f26fce487fdabR43-R52.) to make the behaviour of the sniff a little smarter to check for a few obvious "local usage scenarios", but our usage [here](https://github.com/inpsyde/modularity/blob/bd652971fa0f1874f189930da2eea154cf4d139b/src/Properties/LibraryProperties.php#L35) still causes the warning.

It seems we have 2 options:

* Try to trigger the "local usage detection" somehow. [Judging from the code](https://github.com/WordPress/WordPress-Coding-Standards/blob/dfd6c3a8ee61eed6af15bda09f4a33d79eb2b86b/WordPress/Sniffs/WP/AlternativeFunctionsSniff.php#L259-L310), the only real option is to pass `$use_include_path = true`, but I am not sure we want that
* Accompany the line with a comment like `// phpcs:ignore WordPress.WP.AlternativeFunctions.file_get_contents_file_get_contents -- Reading local file is OK.` in order to disable the sniff. The actual plugin review process may or may not ignore comments like these. But it would at least put up a strong argument that we are aware of the situation and that we are facing a false-positive here.

cc @mmaymo 

### Reproduction instructions

* Download & install a package that uses modularity, for example [Mollie](https://de.wordpress.org/plugins/mollie-payments-for-woocommerce/) which made us aware of the issue
* Download & install [Plugin Check](https://wordpress.org/plugins/plugin-check/)
* Execute tests ( `wp plugin check <PLUGIN_FILE>.php` )

### Expected behavior

In a perfect world, we would see no complaints originating from modularity

### Relevant log output

```shell
FILE: lib/packages/Inpsyde/Modularity/Properties/LibraryProperties.php 43	29	ERROR	WordPress.WP.AlternativeFunctions.file_get_contents_file_get_contents	file_get_contents() is discouraged. Use wp_remote_get() for remote URLs instead.
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by gmazzap** [comment]
I don't consider this a bug.

Every project is expected to comply with the coding standards it commits to, and Modularity is not committed to WordPress coding standards, nor it is committed to "Plugin Check".

This library respects the coding standards it committs to, so there's no unexpected behavior, hence there's no bug.

At the current state, I see this as a problem of the specific project (Mollie, in your example) that decided to use this library and also to be compliant with "Plugin Check".

So normally this is something I would expect to be solved at that project level, but honestly, I'm not sure this is possible. For sure, it is possible to have in that plugin a PHPCS configuration like:

```xml
<rule ref="WordPress.WP.AlternativeFunctions.file_get_contents_file_get_contents">
  <exclude-pattern>*/LibraryProperties.php</exclude-pattern>
</rule>
```

But I have no idea if "Plugin Check" would respect the custom PHPCS configuration in the plugin.

We _could_ commit this library to "Plugin Check" compatibility, but that creates a coupling that could be problematic. Even because we should consider doing the same in other libraries, and possibly checking automatically.

So, to sum it up, I see 2 options:

- We _don't_ commit to respect Plugin Check, so there's nothing to do here.
- We _do_ commit to Plugin Check. But that's a _feature request_ (not a bug) that deserves to be evaluated in a wider context, considerng impact to this and other libraries, look for automatic checks, etc.

Limiting the issue to this very narrow scope of this very specific error report, and so adding a single inline comment to bypass a styling rule the project is not committed to... is not the way I see it done.
